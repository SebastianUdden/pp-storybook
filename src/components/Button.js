import React from "react";
import styled from "styled-components";

const StyledButton = styled.button`
  background-color: ${p => p.backgroundColor || "inherit"};
  padding: 0.5rem;
`;

const Button = ({ children, onClick, backgroundColor }) => (
  <StyledButton onClick={onClick} backgroundColor={backgroundColor}>
    {children}
  </StyledButton>
);

export default Button;
