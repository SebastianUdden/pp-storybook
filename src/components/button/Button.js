import styled from "styled-components";

export const Button = styled.button`
  cursor: ${p => (p.disabled ? "default" : "pointer")};
  text-transform: uppercase;
  color: ${p => p.color || "inherit"};
  min-width: 4rem;
  padding: 0.8rem;
  margin: 0rem 0.2rem;
`;
