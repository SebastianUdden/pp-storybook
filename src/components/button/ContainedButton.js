import React from "react";
import styled from "styled-components";
import {
  BACKGROUND_ACTIVE,
  ON_BACKGROUND,
  DP2,
  DP8,
  SURFACE,
  ON_SURFACE,
  BACKGROUND
} from "../../constants/theme";
import Button from "./Button";
import Ripple from "../ripple/Ripple";

export const StyledButton = styled(Button)`
  background-color: ${p => p.backgroundColor || "inherit"};
  border: none;
  box-shadow: ${p => (!p.disabled ? p.boxShadow : "inherit")};
  transition: box-shadow 0.1s ease-in, background-color ease-in 0.1s;

  :hover {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}DD`
        : p.backgroundColor};
    box-shadow: ${p =>
      !p.disabled ? p.raisedBoxShadow || p.boxShadow : "none"};
  }
  :focus {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}88`
        : "inherit"};
    box-shadow: ${p =>
      !p.disabled ? p.raisedBoxShadow || p.boxShadow : "inherit"};
  }
  :active {
    box-shadow: ${p => (!p.disabled ? p.boxShadow : "inherit")};
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}CC`
        : "inherit"};
  }
`;

const ContainedButton = ({
  children,
  onClick,
  color = ON_BACKGROUND,
  backgroundColor = BACKGROUND,
  backgroundColorHover,
  boxShadow = DP2,
  raisedBoxShadow = DP8,
  disabled
}) => (
  <Ripple onClick={onClick} color={color}>
    <StyledButton
      color={disabled ? `${ON_SURFACE}33` : color}
      backgroundColor={disabled ? SURFACE : backgroundColor}
      backgroundColorHover={
        backgroundColorHover || backgroundColor || BACKGROUND_ACTIVE
      }
      boxShadow={boxShadow}
      raisedBoxShadow={raisedBoxShadow}
      disabled={disabled}
    >
      {children}
    </StyledButton>
  </Ripple>
);

export default ContainedButton;
