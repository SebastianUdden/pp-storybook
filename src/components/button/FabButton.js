import React from "react";
import styled from "styled-components";
import {
  BACKGROUND_ACTIVE,
  ON_BACKGROUND,
  DP6,
  DPraised,
  SURFACE,
  ON_SURFACE,
  BACKGROUND
} from "../../constants/theme";
import { Button } from "./Button";

export const StyledButton = styled(Button)`
  background-color: ${p => p.backgroundColor || "inherit"};
  border-radius: 50%;
  min-width: 3.3rem;
  ${p => p.mini && `padding: 0.4rem; margin: 0; min-width: 2rem;`}
  border: none;
  box-shadow: ${p => (!p.disabled ? p.boxShadow : "inherit")};
  transition: box-shadow 0.1s ease-in, background-color ease-in 0.1s;

  :hover {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}DD`
        : p.backgroundColor};
    box-shadow: ${p =>
      !p.disabled ? p.raisedBoxShadow || p.boxShadow : "none"};
  }
  :focus {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}88`
        : "inherit"};
    box-shadow: ${p =>
      !p.disabled ? p.raisedBoxShadow || p.boxShadow : "inherit"};
  }
  :active {
    box-shadow: ${p => (!p.disabled ? p.boxShadow : "inherit")};
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}CC`
        : "inherit"};
  }
`;

const FabButton = ({
  children,
  onClick,
  color = ON_BACKGROUND,
  backgroundColor = BACKGROUND,
  backgroundColorHover,
  boxShadow = DP6,
  raisedBoxShadow = DPraised,
  mini,
  disabled
}) => (
  <StyledButton
    onClick={onClick}
    color={disabled ? `${ON_SURFACE}33` : color}
    backgroundColor={disabled ? SURFACE : backgroundColor}
    backgroundColorHover={
      backgroundColorHover || backgroundColor || BACKGROUND_ACTIVE
    }
    boxShadow={boxShadow}
    raisedBoxShadow={raisedBoxShadow}
    mini={mini}
    disabled={disabled}
  >
    {children}
  </StyledButton>
);

export default FabButton;
