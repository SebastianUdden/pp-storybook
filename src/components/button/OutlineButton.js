import React from "react";
import styled from "styled-components";
import { BACKGROUND_ACTIVE, ON_BACKGROUND } from "../../constants/theme";
import Button from "./Button";
import Ripple from "../ripple/Ripple";

export const StyledButton = styled(Button)`
  background-color: inherit;
  border: 1px solid ${p => p.backgroundColorHover || "inherit"};

  :hover {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}33`
        : "inherit"};
  }
  :focus {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}88`
        : "inherit"};
  }
  :active {
    background-color: ${p =>
      p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}BB`
        : "inherit"};
  }
`;

const OutlineButton = ({
  children,
  onClick,
  color = ON_BACKGROUND,
  backgroundColorHover = BACKGROUND_ACTIVE,
  disabled
}) => (
  <Ripple onClick={onClick} color={color}>
    <StyledButton
      color={disabled ? backgroundColorHover : color}
      backgroundColorHover={backgroundColorHover}
      disabled={disabled}
    >
      {children}
    </StyledButton>
  </Ripple>
);

export default OutlineButton;
