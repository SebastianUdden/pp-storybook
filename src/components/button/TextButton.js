import React from "react";
import styled from "styled-components";
import { BACKGROUND_ACTIVE, ON_BACKGROUND } from "../../constants/theme";
import { Button } from "./Button";

export const StyledButton = styled(Button)`
  background-color: inherit;
  border: none;
  padding: ${p => (p.small ? "0.3rem" : "inherit")};

  :hover {
    background-color: ${p =>
      p.backgroundColorHover ? `${p.backgroundColorHover}22` : "inherit"};
  }
  :focus {
    background-color: ${p =>
      p.backgroundColorHover ? `${p.backgroundColorHover}88` : "inherit"};
  }
  :active {
    background-color: ${p =>
      p.backgroundColorHover ? `${p.backgroundColorHover}BB` : "inherit"};
  }
`;

const TextButton = ({
  children,
  onClick,
  color = ON_BACKGROUND,
  backgroundColorHover = BACKGROUND_ACTIVE,
  disabled,
  small,
  fontSize,
  fontWeight
}) => (
  <StyledButton
    onClick={onClick}
    color={disabled ? backgroundColorHover : color}
    backgroundColorHover={!disabled && backgroundColorHover}
    disabled={disabled}
    small={small}
    fontSize={fontSize}
    fontWeight={fontWeight}
  >
    {children}
  </StyledButton>
);

export default TextButton;
