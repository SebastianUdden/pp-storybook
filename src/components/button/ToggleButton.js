import React from "react";
import styled from "styled-components";
import {
  BACKGROUND_ACTIVE,
  ON_BACKGROUND,
  BACKGROUND,
  ON_SURFACE
} from "../../constants/theme";
import { Button } from "./Button";

export const StyledButton = styled(Button)`
  margin: 0;
  background-color: ${p => (p.selected ? p.backgroundColor : "inherit")};
  border: 1px solid ${p => p.backgroundColorHover || "inherit"};

  :hover {
    background-color: ${p =>
      p.selected
        ? p.backgroundColor
        : p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}33`
        : "inherit"};
  }
  :focus {
    background-color: ${p =>
      p.selected
        ? p.backgroundColor
        : p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}88`
        : "inherit"};
  }
  :active {
    background-color: ${p =>
      p.selected
        ? p.backgroundColor
        : p.backgroundColorHover && !p.disabled
        ? `${p.backgroundColorHover}BB`
        : "inherit"};
  }
`;

const ToggleButton = ({
  children,
  onClick,
  color = ON_BACKGROUND,
  backgroundColor = BACKGROUND,
  backgroundColorHover = BACKGROUND_ACTIVE,
  disabled,
  selected
}) => (
  <StyledButton
    onClick={onClick}
    color={disabled ? `${ON_SURFACE}33` : color}
    backgroundColor={backgroundColor}
    backgroundColorHover={backgroundColorHover}
    selected={selected}
    disabled={disabled}
  >
    {children}
  </StyledButton>
);

export default ToggleButton;
