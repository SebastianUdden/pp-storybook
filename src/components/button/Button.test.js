import React from "react";
import renderer from "react-test-renderer";
import { action } from "@storybook/addon-actions";
import Button from "./Button";
import TextButton from "./TextButton";
import OutlineButton from "./OutlineButton";
import ToggleButton from "./ToggleButton";
import ContainedButton from "./ContainedButton";
import FabButton from "./FabButton";

describe("Button tests", () => {
  test("Button renders correctly", () => {
    const tree = renderer
      .create(<Button onClick={action("clicked")}>Default Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Text button renders correctly", () => {
    const tree = renderer
      .create(<TextButton onClick={action("clicked")}>Text Button</TextButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Outline button renders correctly", () => {
    const tree = renderer
      .create(
        <OutlineButton onClick={action("clicked")}>
          Outline Button
        </OutlineButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Contained button renders correctly", () => {
    const tree = renderer
      .create(
        <ContainedButton onClick={action("clicked")}>
          Contained Button
        </ContainedButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Toggle button renders correctly", () => {
    const tree = renderer
      .create(
        <ToggleButton onClick={action("clicked")}>Toggle Button</ToggleButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Fab button renders correctly", () => {
    const tree = renderer
      .create(<FabButton onClick={action("clicked")}>Fab Button</FabButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
