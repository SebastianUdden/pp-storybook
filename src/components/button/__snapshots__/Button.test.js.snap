// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Button renders correctly 1`] = `
<button
  className="sc-bdVaJa kOHnq"
  onClick={[Function]}
>
  Default Button
</button>
`;

exports[`Button tests Contained button renders correctly 1`] = `
<button
  className="sc-bdVaJa sc-ifAKCX hmsgCy"
  color="#FFFFFF"
  onClick={[Function]}
>
  Contained Button
</button>
`;

exports[`Button tests Outline button renders correctly 1`] = `
<button
  className="sc-bdVaJa sc-htpNat kNSZir"
  color="#FFFFFF"
  onClick={[Function]}
>
  Outline Button
</button>
`;

exports[`Button tests Text button renders correctly 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH ftLMhZ"
  color="#FFFFFF"
  onClick={[Function]}
>
  Text Button
</button>
`;

exports[`Button tests Toggle button renders correctly 1`] = `
<button
  className="sc-bdVaJa sc-bxivhb dEDINV"
  color="#FFFFFF"
  onClick={[Function]}
>
  Toggle Button
</button>
`;
